rrdd/rrdd_server.ml

```diff
@@ -594,17 +593,17 @@ module Plugin = struct
     (* Read, parse, and combine metrics from all registered plugins. *)
     let read_stats () : (Rrd.ds_owner * Ds.ds) list =
       let plugins = Mutex.execute registered_m
-          (fun _ -> Hashtblext.to_list registered) in
+          (fun _ -> Hashtbl.to_seq registered) in
```

Comment:
I'm not sure whether this is still thread-safe. Previously you would've gotten the full list here.
Now Hashtbl can be mutated while you call the next element of seq. I don't think that to_seq takes a snapshot, its a lazy operation.
In particular see the docs of Hashtbl.to_seq `The behavior is not defined if the hash table is modified during the iteration.`

Reply:
I think you need to make the full conversion to list while holding the mutex, so add a List.of_seq here.

Reply:
Not sure if it's worth keeping the later transformations on Seq instead of List, I've used a list for all the process.